Fuel download large file fails

When I wanna download big file (200MB) I got OOM error
Is there a solution? (I can set buffer in httpUrl android custom downloader, but here I don't know how to do it properly)
Any idea?

Fuel.download(url)
                .fileDestination { _, _ ->
                    File(path, fileName)
                }
                .progress { readBytes, totalBytes ->

                    activity.runOnUiThread {

                        binding.tvBookSizeInMegabyte.text =
                                context.getString(R.string.byte_to_mega_byte ,
                                                  utils.bytesToMBFormatWithoutLetters(readBytes) ,
                                                  utils.bytesToMBFormatWithoutLetters(totalBytes)
                                )
                    }
                }
                .timeout(8000)
                .response { request, response, result ->

                    result.fold(

                            success = {

                                when (type) {
                                    Constants.DownloadType.AR -> {
                                        downloadARCompleted(fileName)
                                    }
                                    Constants.DownloadType.Ebook -> {
                                        handleDownloadStatus(BookDownloadStatus.Downloaded)
                                    }
                                }

                            },
                            failure = {

                                handleDownloadStatus(BookDownloadStatus.DownloadFailed)

                                when (type) {

                                    Constants.DownloadType.AR -> File(utils.purchasePath(context) + fileName).delete()
                                    Constants.DownloadType.Ebook -> File(utils.EBookPath(context) + fileName).delete()

                                }

                                if (!it.causedByInterruption) {

                                
                                    if (response.statusCode == 404 && response.responseMessage == "Not Found") {

                                        CustomToast().showToast(
                                                context,
                                                activity,
                                                dialogContainer,
                                                context.getString(R.string.nothing_to_download),
                                                Toast.LENGTH_SHORT,
                                                Gravity.BOTTOM
                                        )
                                    }
                                }

                            }
                    )
                }

        }catch (e : Exception){
            handleException(e)
        }

******************************************************************************************

This is the code snippet for downloading files, I get this error (I think it depends on android device ram in my test case it has 3GB memory)

******************************************************************************************

I put error log here:

Failed to allocate a 134217744 byte allocation with 25165824 free bytes and 108MB until OOM, target footprint 112943576, growth limit 201326592
    	com.github.kittinunf.fuel.core.FuelError$Companion.wrap(FuelError.kt:86)
    	com.github.kittinunf.fuel.core.FuelError$Companion.wrap$default(FuelError.kt:83)
    	com.github.kittinunf.fuel.core.requests.RequestTaskCallbacks.call(RequestTaskCallbacks.kt:30)
    Caused by: java.lang.OutOfMemoryError: Failed to allocate a 134217744 byte allocation with 25165824 free bytes and 108MB until OOM, target footprint 112943576, growth limit 201326592
    	java.util.Arrays.copyOf(Arrays.java:3161)
    	java.io.ByteArrayOutputStream.grow(ByteArrayOutputStream.java:118)
    	java.io.ByteArrayOutputStream.ensureCapacity(ByteArrayOutputStream.java:93)
    	java.io.ByteArrayOutputStream.write(ByteArrayOutputStream.java:153)
    	kotlin.io.ByteStreamsKt.copyTo(IOStreams.kt:108)
    	kotlin.io.ByteStreamsKt.copyTo$default(IOStreams.kt:103)
    	com.github.kittinunf.fuel.core.requests.DefaultBody.writeTo(DefaultBody.kt:87)
    	com.github.kittinunf.fuel.core.requests.DefaultBody.toByteArray(DefaultBody.kt:50)
    	com.github.kittinunf.fuel.core.Response.getData(Response.kt:32)

--- a/fuel/README.md
+++ b/fuel/README.md
@@ -596,6 +596,32 @@ The `stream` variant expects your callback to provide a `Pair` with both the `Ou
   - use an `EmptyDeserializer` with `await(deserializer)` or one of the `response(deserializer)` variants
   - provide an `InputStream` callback that `throws` or returns an empty `InputStream`.
 
+For downloading larger files it is good to use `request.streamDestination { }` instead of `request.fileDestination { }`
+
+```kotlin
+val outputStream = FileOutputStream(File(filePath))
+
+Fuel.download("https://httpbin.org/bytes/32768")
+    .streamDestination { response, request -> Pair(outputStream, { request.body.toStream() }) }
+    .progress { readBytes, totalBytes ->
+        val progress = readBytes.toFloat() / totalBytes.toFloat() * 100
+        println("Bytes downloaded $readBytes / $totalBytes ($progress %)")
+    }
+    .response { result -> 
+
+        result.fold(
+
+                success = {
+                    
+                },
+
+                failure = {
+                    
+                }
+        )
+    }
+```
+
 ### Cancel an async `Request`
 
 The `response` functions called with a `handler` are async and return a `CancellableRequest`. These requests expose a few extra functions that can be used to control the `Future` that should resolve a response:

